### DISCUSSION TITLE
Configuring Emacs as a PHP IDE - LSP Mode - LSP support for Emacs

### DISCUSSION THREAD

**Discussion by giscus** [emacs-lsp/lsp-mode#3777] [Giscus]
# Configuring Emacs as a PHP IDE - LSP Mode - LSP support for Emacs

Language Server Protocol Support for Emacs

https://emacs-lsp.github.io/lsp-mode/tutorials/php-guide/

**Comment by phisu** [comment]
i have successfully installed and configured xdebug. and now i am trying to get debugging to run. in test.php i set a breakpoint as described in the example and do M-x dap-debug and select "PHP Debug" from the list. but then i get an error message in the minibuffer. 

Messages shows me the following error:

Debug session process exited with status: exited abnormally with code 1
error in process sentinel: Symbol's function definition is void: dap-hydra/nil

and in the buffer PHP Debug stderr I find: 

node:internal/modules/cjs/loader:959
  throw err;
  ^

Error: Cannot find module '/home/sutter/.emacs.d/.extension/vscode/xdebug.php-debug/extension/out/phpDebug.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
    at Function.Module._load (node:internal/modules/cjs/loader:804:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

i don't understand the error message. i can find the directory .emacs.d/.extension/vscode/xdebug.php-debug but it is empty. did i miss something to install? or what do the error messages want to tell me?

**Reply by yyoncho** [reply]
Most likely you have to download manually the php extension and expand it manually. https://github.com/xdebug/vscode-php-debug/releases

**Reply by phisu** [reply]
i downloaded  
https://github.com/xdebug/vscode-php-debug/archive/refs/tags/v1.29.0.zip 

and unzipped it to the ~/.emacs.d/.extension/vscode directory. 

but i still get the same errors. did i get "expand" wrong?

**Reply by yyoncho** [reply]
is there anything on this path /home/sutter/.emacs.d/.extension/vscode/xdebug.php-debug/extension/out/phpDebug.js ?

**Reply by phisu** [reply]
/home/sutter/.emacs.d/.extension/vscode/xdebug.php-debug was empty and I can't imagine how to install anything into it.

**Comment by yyoncho** [comment]
Use https://github.com/xdebug/vscode-php-debug/releases/download/v1.29.0/php-debug-1.29.0.vsix  It has `extension/out/phpDebug.js`

**Reply by phisu** [reply]
Thank you!  This helped! Now I have to learn how dap-mode is to use.
